src/ranch_ssl.erl

```diff
@@ -296,3 +297,18 @@ cleanup(#{socket_opts:=SocketOpts}) ->
 	end;
 cleanup(_) ->
 	ok.
+
+-spec strip_usupported_options(opts()) -> opts().
+strip_usupported_options(SocketOpts) ->
+    case lists:keyfind(versions, 1, SocketOpts) of
+        {versions, ['tlsv1.3']} ->
+            Intermediate1 = lists:keydelete(secure_renegotiate, 1, SocketOpts),
+            Intermediate2 = lists:keydelete(reuse_sessions, 1, Intermediate1),
+            Intermediate3 = lists:keydelete(next_protocols_advertised, 1, Intermediate2),
+            lists:keydelete(alpn_preferred_protocols, 1, Intermediate3);
```

Comment:
Minor detail probably, but this is not exhaustive either. `lists:keydelete` only removes the _first_ occurrence, but in reality the same option may be given multiple times (if it follows the practice used in `gen_tcp`, not 100% sure for `ssl`). To be on the safe side, you'll have to `lists:filter/2` the socket options.

Reply:
> Minor detail probably, but this is not exhaustive either. `lists:keydelete` only removes the _first_ occurrence, but in reality the same option may be given multiple times, pointless as that may be (if it follows the practice used in `gen_tcp`, not 100% sure for `ssl`). To be on the safe side, you'll have to `lists:filter/2` the socket options.